### PR TITLE
JNG-4297 Creating Judo test methodology

### PR DESCRIPTION
**Pull request by fmagnucz**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-4297" title="JNG-4297" target="_blank">StoryJNG-4297</a>  Creating Judo test methodology
  </td></table>
  <br />
 

JNG-4297 Creating Judo test methodology

### CODE REVIEW

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/CompositionRelationshipsTest.java [hunk 1]
```diff
@@ -238,6 +299,12 @@ void testManualTransactionManagementCommit() throws SystemException, NotSupporte
 
     @Test
     @Disabled
+    @Requirement(reqs = {"REQ-TYPE-001",
```
Comment (bencegelei) [inline judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/CompositionRelationshipsTest.java]: weird indentation

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/CompositionRelationshipsTest.java [hunk 2]
```diff
@@ -253,6 +320,12 @@ void testMultipleInheritance() {
     }
 
     @Test
+    @Requirement(reqs = {"REQ-TYPE-001",
```
Comment (bencegelei) [inline judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/CompositionRelationshipsTest.java]: weird indentation

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/PrimitiveFunctionsTest.java [hunk 3]
```diff
@@ -27,6 +27,7 @@
 import hu.blackbelt.judo.runtime.core.jsl.itest.primitives.sdk.primitives.primitives.MyEntityWithOptionalFields;
 import hu.blackbelt.judo.runtime.core.jsl.itest.primitives.sdk.primitives.primitives.MyEnum;
 import hu.blackbelt.judo.sdk.query.StringFilter;
+import hu.blackbelt.judo.test.Requirement;
```
Comment (bencegelei) [inline judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/PrimitiveFunctionsTest.java]: Requirements are missing in this file

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/test/TestDefinition.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+ */
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention( RetentionPolicy.SOURCE )
+@Target( value = { ElementType.METHOD, ElementType.TYPE })
+public @interface TestDefinition {
```
Comment (bencegelei) [inline judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/test/TestDefinition.java]: I did not find any usage of this annotation in this PR's scope

**Review by bencegelei — CHANGES_REQUESTED** [review]

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/CompositionRelationshipsTest.java (hunk 1 above)
Comment (fmagnucz) [reply judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/CompositionRelationshipsTest.java]: Thank you!

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/CompositionRelationshipsTest.java (hunk 2 above)
Comment (fmagnucz) [reply judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/CompositionRelationshipsTest.java]: Thank you!

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/PrimitiveFunctionsTest.java (hunk 3 above)
Comment (fmagnucz) [reply judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/PrimitiveFunctionsTest.java]: Thank you. I have deleted it.

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/test/TestDefinition.java (hunk 4 above)
Comment (fmagnucz) [reply judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/test/TestDefinition.java]: Yes, this is not used yet, but I have deleted.

**Review by robertcsakany — APPROVED** [review]

**Review by bencegelei — APPROVED** [review]

**Pull request merged**